pyseext/js/PySeExt.GridHelper.js

```diff
@@ -61,53 +61,44 @@ globalThis.PySeExt.GridHelper = {
      * @return {Element} The element for the found row.
      */
     getRow: function(gridSelector, rowData) {
-        var row,
-            grids = globalThis.Ext.ComponentQuery.query(gridSelector),
-            grid,
-            store,
-            foundIndex,
-            prop;
+        var me = this,
+        grids = globalThis.Ext.ComponentQuery.query(gridSelector),
+        grid,
+        store;
 
         if (grids && grids.length) {
             grid = grids[0];
             store = grid.getStore();
 
-            if (typeof(rowData) === 'number') {
-                foundIndex = rowData;
-
-                // Verify it's not bollocks
-                if (!Ext.isNumber(parseInt(foundIndex)) || foundIndex < 0) {
-                    globalThis.Ext.raise('If attempting to get a row for a specific index, the index must be an integer greater than or equal to zero.');
-                }
+            row = me.__findRowRecord(rowData, store);
+        }
 
-                // Check that found index is within bounds, if so grab the row.
-                if (foundIndex < store.getCount()) {
-                    row = grid.getView().getRow(foundIndex);
-                }
-            } else {
-                foundIndex = store.findBy(function(record, id) {
-                    var hasRecordBeenFound = true;
-
-                    for (prop in rowData) {
-                        if (rowData.hasOwnProperty(prop)) {
-                            if (record.get(prop) !== rowData[prop]) {
-                                hasRecordBeenFound = false;
-                                break;
-                            }
-                        }
-                    }
+        return grid.getView().getRow(row);
+    },
+    /**
+     * Gets the row store with the specified data or index in the grid.
+     * The row is scrolled into view ready for clicking by the caller.
+     *
+     * The grid must be visible.
+     *
+     * @param  {String}        gridSelector The selector for the grid.
+     * @param  {Object|Number} rowData      The index of or an object containing the row data for the record to be found.
+     * @return {store} The element for the found row.
+     */
+        getRowStore: function(gridSelector, rowData) {
+            var me = this,
+                grids = globalThis.Ext.ComponentQuery.query(gridSelector),
+                grid,
+                store;
 
-                    return hasRecordBeenFound;
-                });
+                if (grids && grids.length) {
+                    grid = grids[0];
+                    store = grid.getStore();
 
-                if (foundIndex !== -1) {
-                    row = grid.getView().getRow(foundIndex);
```

Comment:
The variable 'row' has not been declared meaning this creates a global!

Reply:
have now declared